system/Commands/Utilities/Routes/FilterFinder.php

```diff
@@ -50,6 +51,12 @@ public function find(string $uri): array
     {
         $this->filters->reset();
 
+        $isLocaleExist = strpos($uri, '{locale}') !== false;
```

Comment:
I need help. How do I get the current Routes set in the test? 
```php
    public function testFindFiltersWithSupportedLocalesOnly(): void
    {
        $collection = $this->createRouteCollection();
        $collection->useSupportedLocalesOnly(true); // changed
        $collection->get('{locale}/admin/(:segment)', ' AdminController::index/$1');
        $router  = $this->createRouter($collection);
        $filters = $this->createFilters();
        $finder = new FilterFinder($router, $filters);
        // d(
            // $router, protected 'useSupportedLocalesOnly' -> boolean true
            // $finder, protected 'useSupportedLocalesOnly' -> boolean true
        // );

        // But in find()
        //  if (strpos($uri, '{locale}') !== false && Services::routes()->shouldUseSupportedLocalesOnly()) ...
        // Services::routes() return from app/Config. Can't get from the $this->router
        $filters = $finder->find('{locale}/admin/settings');

        $expected = [
            'before' => ['!csrf'],
            'after'  => ['!toolbar'],
        ];
        $this->assertSame($expected, $filters);
    }
```

Reply:
Inject the `$collection` into `Services` with `injectMock()`.
See https://codeigniter4.github.io/CodeIgniter4/testing/overview.html#services-injectmock

Don't forget to reset it in `Services` after the test.

Reply:
>         // But in find()
>        //  if (strpos($uri, '{locale}') !== false && Services::routes()->shouldUseSupportedLocalesOnly()) ...
>        // Services::routes() return from app/Config. Can't get from the $this->router

Where is the code? Why do you need to call `Services::routes()`?

Reply:
Thanks!